api/product.rb

```diff
@@ -39,17 +40,27 @@ def validate
 
     # Represents a version of the API for this product
     class Version < Api::Object
+      include Comparable
       attr_reader :base_url
       attr_reader :default
       attr_reader :name
 
+      ORDER = %w[v1 beta alpha].freeze
```

Comment:
If we take the sql instance version naming scheme, would you call `v1beta1` version only `beta` in Magic Modules?

What about the v1beta1 and v1beta2? Would we only support one of these two?

To support arbitrary naming scheme, we could derive this order from the order in `versions` list in api.yaml.

Reply:
Just sent some emails to get more information from API teams. My initial thoughts were that "v1" (which I should rename to `ga` to align better with https://cloud.google.com/terms/launch-stages) would be the main one, and then if there's a separate beta or alpha it would get listed. Having only certain possible names will make it easier for us to generate Terraform versions, so we don't accidentally end up with terraform-provider-google, terraform-provider-google-v1beta1, terraform-provider-google-v0.beta, terraform-provider-google-v2beta1, etc. Does that seem reasonable?

Reply:
Yup, it makes total sense. For now we can proceed with this approach and change later if needed after we get more information from the API teams.